### ISSUE TITLE
训练时一直在Training Progress开始卡住，但是gpu是被占用的

### ISSUE THREAD

**Issue by SimonHeye** [volcengine/verl#3464]
训练时一直在Training Progress开始卡住，但是gpu是被占用的，且没有出现OOM的错误，只显示：Training Progress:   0%|          | 0/125 [04:45<?, ?it/s]

**Comment by denake-on** [comment]
您好 我也是这个问题请问您解决了吗

**Comment by Kirrito-k423** [comment]
中文：

感谢反馈。目前公开信息只有 `Training Progress` 0% 卡住、GPU 被占用且没有 OOM，这还不足以判断根因。`Training Progress` 进度条是在 PPO 训练循环里创建的，并且只有一个完整 training step 结束后才会前进；第一步内部会依次涉及 rollout generation、reward、old/ref logprob、critic/actor update、weight sync 等阶段，所以仅凭这一行无法区分是 rollout、reward/custom function、Ray worker、NCCL 通信还是数据/配置问题。

请补充以下信息，方便继续判断：

1. 完整启动命令和所有 config overrides。
2. verl commit/version，以及 PyTorch/CUDA/NCCL/driver/Ray/vLLM 或 SGLang 版本。
3. GPU 型号、数量、`nvidia-smi topo -m` 输出，以及单卡和多卡是否表现不同。
4. 卡住前后的 driver 日志和 Ray worker 日志，尤其是 `/tmp/ray/session_latest/logs` 下 TaskRunner/WorkerDict 相关日志。建议复现时先设置：

```bash
export HYDRA_FULL_ERROR=1
export RAY_DEDUP_LOGS=0
export NCCL_DEBUG=INFO
```

5. 是否启用了 `custom_reward_function`、reward model、`trainer.val_before_train`，以及使用的是 vLLM 还是 SGLang rollout。
6. 对比尝试：`export NCCL_P2P_DISABLE=1`、减小 batch/rollout.n/log_prob_micro_batch_size_per_gpu、禁用自定义 reward 或 validation 后是否仍卡在同一位置。

如果进程仍卡住，也请对 TaskRunner/WorkerDict 的 PID 采集一次 Python 栈，例如 `pystack remote $PID`，这样更容易判断实际停在哪个阶段。

English:

Thanks for the report. The current public information only says that training stays at `Training Progress` 0%, GPU memory is occupied, and there is no OOM. That is not enough to identify the root cause. In the PPO training loop, the `Training Progress` bar advances only after a full training step finishes. The first step may include rollout generation, reward computation, old/ref logprob computation, critic/actor update, and weight synchronization, so this single line cannot distinguish rollout, reward/custom code, Ray worker, NCCL communication, or data/config issues.

Please share the full launch command and config overrides, verl commit/version, PyTorch/CUDA/NCCL/driver/Ray/vLLM or SGLang versions, GPU model/count, `nvidia-smi topo -m`, whether single-GPU and multi-GPU behave differently, and full driver plus Ray worker logs around the hang, especially under `/tmp/ray/session_latest/logs`. For the next reproduction, please enable:

```bash
export HYDRA_FULL_ERROR=1
export RAY_DEDUP_LOGS=0
export NCCL_DEBUG=INFO
```

It would also help to state whether `custom_reward_function`, a reward model, `trainer.val_before_train`, and vLLM/SGLang rollout are enabled. Please compare whether the hang still happens after trying `export NCCL_P2P_DISABLE=1`, reducing batch/rollout.n/log_prob_micro_batch_size_per_gpu, and temporarily disabling custom reward or validation if applicable.

If the process is still hanging, please capture a Python stack from the TaskRunner/WorkerDict PID, for example with `pystack remote $PID`, so maintainers can see which stage is actually blocked.

This response was generated by a scheduled Codex GPT5.5 issue processing task.
